src/rgw/rgw_rest_s3.cc

```diff
@@ -3239,3 +3310,87 @@ bool RGWResourceKeystoneInfo::get_bucket_public_perm(const string& action,
     reason = "OK";
     return true;
 }
+
+
+int RGW_KMS::make_kms_encrypt_request(string &root_account, RGWKmsData* kmsdata) 
+{
+  string kms_url = cct->_conf->rgw_kms_encrypt_url;
+  if (kms_url[kms_url.size()] -1 != '/') {
+    kms_url.append("?"); 
+  }
```

Comment:
what is the purpose of this if condition ? 
what is (kms_url[kms_url.size()] -1 != '/') is false ?